### PR TITLE
improvement(NotificationDrawer): Add x close button to notification drawer

### LINKED ISSUE #4201: NotificationDrawer: Add an 'x' to the notification drawer so the user can easily close
**Describe the issue. What is the expected and unexpected behavior?**
Add an 'x' to the notification drawer so the user can easily close
Opened issue in core here: https://github.com/patternfly/patternfly/issues/3072

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
This is a bug on openshift: https://github.com/patternfly/patternfly/issues/3072

### PR DESCRIPTION
**Pull request by jcaianirh**
Add x close button to the notification drawer, and update the integration and md demos to reflect
the change. Also, add an integration test.

**What**: Closes #4201 

Screen Shot 2020-08-26 at 5 26 45 PM

### CODE REVIEW

File: packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx [hunk 1]
```diff
@@ -5,28 +5,53 @@ import styles from '@patternfly/react-styles/css/components/NotificationDrawer/n
...
-  /**  Notification drawer heading custom text which can be used instead of providing count/unreadText */
-  customText?: string;
 }
 
+const NotificationDrawerCloseButton: React.FunctionComponent<NotificationDrawerCloseButtonProps> = ({
+  icon,
+  onClose
+}: NotificationDrawerCloseButtonProps) => (
+  <div>
+    <button className="pf-c-button pf-m-plain" type="button" aria-label="Close" onClick={onClose}>
```
Comment (tlabaj) [inline packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx:39]: the aria-label needs to be customizable for localization.

File: packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx [hunk 2]
```diff
@@ -5,28 +5,53 @@ import styles from '@patternfly/react-styles/css/components/NotificationDrawer/n
...
+  onClose: () => void;
+}
+
 export interface NotificationDrawerHeaderProps extends React.HTMLProps<HTMLDivElement> {
   /**  Content rendered inside the drawer */
   children?: React.ReactNode;
   /**  Additional classes for notification drawer header. */
   className?: string;
+  /** Optional icon other than x */
+  closeIcon?: React.ReactNode;
```
Comment (tlabaj) [inline packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx:21]: @mcarrano do we want the icon to be customizable?

File: packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx [hunk 3]
```diff
@@ -5,28 +5,53 @@ import styles from '@patternfly/react-styles/css/components/NotificationDrawer/n
...
+  onClose?: () => void;
   /**  Notification drawer heading title */
   title?: string;
   /**  Notification drawer heading unread text used in combination with a count */
   unreadText?: string;
-  /**  Notification drawer heading custom text which can be used instead of providing count/unreadText */
-  customText?: string;
 }
 
+const NotificationDrawerCloseButton: React.FunctionComponent<NotificationDrawerCloseButtonProps> = ({
```
Comment (tlabaj) [inline packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx:34]: to follow what we do with our other components, I would suggest moving this class into it's own file.

File: packages/react-integration/demo-app-ts/src/components/demos/NotificationDrawerDemo/NotificationDrawerGroupsDemo.tsx [hunk 4]
```diff
@@ -88,7 +96,7 @@ export class GroupsNotificationDrawerDemo extends React.Component<
     ];
     return (
       <NotificationDrawer>
-        <NotificationDrawerHeader count={4}>
+        <NotificationDrawerHeader count={4} onClose={this.onDrawerClose}>
```
Comment (tlabaj) [inline packages/react-integration/demo-app-ts/src/components/demos/NotificationDrawerDemo/NotificationDrawerGroupsDemo.tsx:99]: Can you also add the other new `closeIcon` prop to the  demo app.

File: packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx (hunk 1 above)
Comment (jcaianirh) [reply packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx:39]: Added custom aria label prop for the button.

File: packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx (hunk 3 above)
Comment (jcaianirh) [reply packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx:34]: sure, exported this to its own file.

File: packages/react-integration/demo-app-ts/src/components/demos/NotificationDrawerDemo/NotificationDrawerGroupsDemo.tsx (hunk 4 above)
Comment (jcaianirh) [reply packages/react-integration/demo-app-ts/src/components/demos/NotificationDrawerDemo/NotificationDrawerGroupsDemo.tsx:99]: Added closeIcon to the groups demo integration app.

**Comment by jcaianirh** [comment]
> Looks good @jcaianirh except I don't understand why the examples here do not show it. Only the demo https://patternfly-react-pr-4739.surge.sh/documentation/react/components/notificationdrawer
I forgot to add this to the function name. :)
Updated to resolve the .md example

**Review by nicolethoen — DISMISSED** [review]

File: packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeaderCloseButton.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  onClose: () => void;
+}
+
+export const NotificationDrawerCloseButton: React.FunctionComponent<NotificationDrawerCloseButtonProps> = ({
+  closeButtonAriaLabel = 'Close',
+  icon,
+  onClose
+}: NotificationDrawerCloseButtonProps) => (
+  <div>
+    <button className="pf-c-button pf-m-plain" type="button" aria-label={closeButtonAriaLabel} onClick={onClose}>
```
Comment (mcoker) [inline packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeaderCloseButton.tsx:18]: just curious, why wouldn't you use the react `<Button>` component?

**Review by mcoker — DISMISSED** [review]
👍 thanks @jcaianirh!

File: packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeaderCloseButton.tsx (hunk 5 above)
Comment (jcaianirh) [reply packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeaderCloseButton.tsx:18]: @mcoker i should, huh?

File: packages/react-integration/cypress/integration/notificationdrawerbasic.spec.ts [hunk 6]
```diff
@@ -5,6 +5,10 @@ describe('Notification Drawer Basic Demo Test', () => {
     cy.url().should('eq', 'http://localhost:3000/notification-drawer-basic-demo-nav-link');
   });
 
+  it('Verify x icon in header', () => {
```
Comment (tlabaj) [inline packages/react-integration/cypress/integration/notificationdrawerbasic.spec.ts:8]: Can you also verify the interaction of closing the drawer?  I noticed the Example in the docs does not seem to be working.

Comment (jcaianirh) [reply packages/react-integration/cypress/integration/notificationdrawerbasic.spec.ts:8]: @tlabaj we don't open/close the drawer in the examples...only in the demo

**Comment by mcarrano** [comment]
@jcaianirh noticed that there is a prop to pass in a different icon for Close. Any reason to allow this to be customized? While I'm all for allowing flexibility, seems like this is something we'd want to keep standard. Thoughts?

**Comment by jcaianirh** [comment]
@mcarrano , really its just about adding flexibility...trying to shortcut a future request to allow flexibility for this icon

**Comment by mcarrano** [comment]
@jcaianirh Generally I would agree with you. But in this case, I don't think I'd want that action to be used for anything other than closing the drawer and that we want to use the close icon consistently. After discussing with @tlabaj we agreed that the prop should be removed. Thanks.

**Comment by jcaianirh** [comment]
@mcarrano will remove it.

**Review by tlabaj — APPROVED** [review]
LGTM

File: packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx [hunk 7]
```diff
@@ -36,7 +44,23 @@ export const NotificationDrawerHeader: React.FunctionComponent<NotificationDrawe
...
       <span className={css(styles.notificationDrawerHeaderStatus)}>{customText || `${count} ${unreadText}`}</span>
     )}
-    {children && <div className={css(styles.notificationDrawerHeaderAction)}>{children}</div>}
+    {children && (
+      <div className={css(styles.notificationDrawerHeaderAction)}>
+        {children}
+        {onClose && (
+          <div>
+            <Button
+              className="pf-c-button pf-m-plain"
```
Comment (tlabaj) [inline packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx:53]: You should not need these classes here. The `Button` already has the `pf-c-button` class. You can just set `variant` to plain to get the `pf-m-plain` modifier applied.

**Review by tlabaj — CHANGES_REQUESTED** [review]

**Review by mcarrano — DISMISSED** [review]
Looks good. Thanks @jcaianirh !

**Review by tlabaj — DISMISSED** [review]
LGTM

**Review by mcarrano — DISMISSED** [review]
Looks good!

**Review by tlabaj — APPROVED** [review]

**Review by mcoker — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Pull request merged**